server/shared/src/main/scala/org/http4s/server/middleware/CSRF.scala

```diff
@@ -352,7 +387,13 @@ object CSRF {
 
     def withHeaderName(headerName: CIString): CSRFBuilder[F, G] =
       copy(headerName = headerName)
-    def withClock(clock: Clock): CSRFBuilder[F, G] = copy(clock = clock)
+
+    @deprecated(
+      "This method doesn't change anything since the `java.time.Clock` has been purged from CSRFBuilder",
+      "v1.0.0-M40",
+    )
+    def withClock(clock: Clock): CSRFBuilder[F, G] = this
```

Comment:
I'm worrying about this breaking behaviour, but I didn't come up with anything better than that. At least, users will be get notified by the compiler. 🤷🏻

Reply:
I think it's a fine heads up.